addons/purchase_stock/wizard/product_replenish.py

```diff
@@ -85,7 +85,7 @@ def _get_replenishment_order_notification_link(self, order_line):
 
     def _get_date_planned(self, route_id, **kwargs):
         date = super()._get_date_planned(route_id, **kwargs)
-        if route_id.name != 'Buy':
+        if route_id.with_context(lang='en_US').name != 'Buy':
```

Comment:
same problem, using name may not always work
```suggestion
        if 'buy' not in route_id.rule_ids.mapped('action'):
```